### ISSUE TITLE
BigDecimal on the left while coercing doesn't have cases for multiplication and division

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1598]
I was surprised by this. I'd though this had been solved by #1533. Unfortunately, not yet for multiplication and division. I noticed when one test in Sequel tests failed in the latest TruffleRuby 1.0.0 RC 13. Ironically, on the same issue reported in #1507 => `BigDecimal('1.0') * Sequel[:a][:y]`

As an example to reproduce the issue, I use @janko's original code in #1507

```ruby
require "bigdecimal"

class MyNumber
  def initialize(number)
    @number = number
  end

  def coerce(other)
    puts "called with #{other.inspect}"

    case other
    when Integer    then [other, @number]
    when BigDecimal then [other, BigDecimal.new(@number)]
    end
  end
end

1                 / MyNumber.new(2)
BigDecimal.new(1) / MyNumber.new(2)

1                 * MyNumber.new(2)
BigDecimal.new(1) * MyNumber.new(2)
```

Error:

```
bigdecimal.rb:19:in `*': Truffle doesn't have a case for the org.truffleruby.stdlib.bigdecimal.BigDecimalNodesFactory$MultOpNodeFactory$MultOpNodeGen node with values of type  BigDecimal(com.oracle.truffle.object.basic.DynamicObjectBasic) MyNumber(com.oracle.truffle.object.basic.DynamicObjectBasic) (TypeError)
	from bigdecimal.rb:19:in `<main>'
```

```
bigdecimal.rb:19:in `/': Truffle doesn't have a case for the org.truffleruby.stdlib.bigdecimal.BigDecimalNodesFactory$DivOpNodeFactory$DivOpNodeGen node with values of type  BigDecimal(com.oracle.truffle.object.basic.DynamicObjectBasic) MyNumber(com.oracle.truffle.object.basic.DynamicObjectBasic) (TypeError)
	from bigdecimal.rb:19:in `<main>'
```

**Comment by eregon** [comment]
Thanks for the report.
Sorry about that, we planned to fix other operations right after integrating #1533 (for which I didn't want to ask the author too much for a first contribution), but due to the OCA process which took much longer than expected and other more urgent tasks, this was forgotten.
I'll fix it.

**Comment by eregon** [comment]
I fixed this in 89e5a01af333182eb221061bac6db3df1e70d399, it will be in the next release.
I checked and the above example works identically on MRI and TruffleRuby now.
Thanks for the report!
